sdk/tables/azure-data-tables/inc/azure/data/tables/credentials/shared_key_credential.hpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <azure/core/http/http.hpp>
+
+#include <memory>
+#include <mutex>
+#include <string>
+namespace Azure { namespace Data { namespace Tables { namespace _internal { namespace Policies {
+  class SharedKeyPolicy;
+  class SharedKeyLitePolicy;
+}}}}} // namespace Azure::Data::Tables::_internal::Policies
+namespace Azure { namespace Data { namespace Tables { namespace Credentials {
+
+namespace Sas {
+  struct AccountSasBuilder;
+} // namespace Sas
+
+/**
+ * @brief A SharedKeyCredential is a credential backed by an account's name and
+ * one of its access keys.
+ */
+class SharedKeyCredential final {
+public:
+  /**
+   * @brief Initializes a new instance of the SharedKeyCredential.
+   *
+   * @param accountName Name of the  account.
+   * @param accountKey Access key of the
+   * account.
+   */
+  explicit SharedKeyCredential(std::string accountName, std::string accountKey)
+      : AccountName{std::move(accountName)}, m_accountKey{std::move(accountKey)}
+  {
+  }
+
+  /**
+   * @brief Update the  account's access key. This intended to be used when you've
+   * regenerated your account's access keys and want to update long lived clients.
+   *
+   * @param accountKey An  account access key.
+   */
+  void Update(std::string accountKey)
+  {
+    std::lock_guard<std::mutex> guard(m_mutex);
+    m_accountKey = std::move(accountKey);
+  }
+
+  /**
+   * @brief Gets the name of the Account.
```

Comment:
```suggestion
   * @brief The name of the Account.
```

A major nit: Because this is *not* a method, it cannot take any action (like "Get").

Reply:
done